pkg/detectors/notion/notion.go

```diff
@@ -22,13 +22,13 @@ var (
 	client = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat = regexp.MustCompile(`\b(secret_[A-Za-z0-9]{43})\b`)
+	keyPat = regexp.MustCompile(`\b((?:secret_[A-Za-z0-9]{43})|ntn_[0-9]{11}[A-Za-z0-9]{32}[A-Za-z0-9]{3})\b`)
```

Comment:
Great 😃 Could you move the new pattern to a separate V2 detector instead of merging it into the existing one? Since the API for verification is the same, we can extract the verification logic into a shared method and use it in both versions.